### PR TITLE
fix(solution): fix migrate v1 project programming language detection

### PR DESCRIPTION
**Pull request by XiaofuHuang**
The origin project has been moved to the archive folder, change the language detection file path

### CODE REVIEW

**Review by swatDong — APPROVED** [review]
The fix logic looks good for the issue.
But considering more, could the sequence be "prepare all required config -> move content -> generate config", in case any error on the preparation step and user's project will not be changed.

**Comment by XiaofuHuang** [comment]
> The fix logic looks good for the issue.
> But considering more, could the sequence be "prepare all required config -> move content -> generate config", in case any error on the preparation step and user's project will not be changed.

Now the create basic folder structure logic is in `core`. So the archive logic is in `core` too. The archive will be executed after all the `core` configuration is prepared but before any solution and resource plugins migration logic.

File: packages/fx-core/src/common/fileUtils.ts [hunk 1]
```diff
@@ -34,3 +34,12 @@ export async function readJson(filePath: string): Promise<any> {
   sendTelemetryErrorEvent(Component.core, TelemetryEvent.ReadJson, fxError);
   throw fxError;
 }
+
+export async function checkFileExist(filePath: string): Promise<boolean> {
```
Comment (1yefuwang1) [inline packages/fx-core/src/common/fileUtils.ts:38]: Does fs.pathExists meet your requirement?

**Review by 1yefuwang1 — COMMENTED** [review]
Could you please add a ut case?

**Comment by XiaofuHuang** [comment]
> Could you please add a ut case?

**Review by 1yefuwang1 — APPROVED** [review]

**Pull request merged**